domain/storageprovisioning/state/volume.go

```diff
@@ -401,6 +401,71 @@ AND    net_node_uuid = $netNodeUUID.uuid
 	return domainstorageprovisioning.VolumeAttachmentUUID(dbVal.UUID), nil
 }
 
+// GetVolumeAttachment returns the volume attachment for the supplied volume
+// attachment uuid.
+//
+// The following errors may be returned:
+// - [storageprovisioningerrors.VolumeAttachmentNotFound] when no volume
+// attachment exists for the provided uuid.
+func (st *State) GetVolumeAttachment(
+	ctx context.Context,
+	uuid domainstorageprovisioning.VolumeAttachmentUUID,
+) (domainstorageprovisioning.VolumeAttachment, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return domainstorageprovisioning.VolumeAttachment{}, errors.Capture(err)
+	}
+
+	var (
+		uuidInput = volumeAttachmentUUID{UUID: uuid.String()}
+	)
+
+	stmt, err := st.Prepare(`
+SELECT &volumeAttachment.* FROM (
+    SELECT DISTINCT sv.volume_id,
+                    sva.life_id,
+                    sva.read_only,
+                    bd.name AS block_device_name,
+                    bd.bus_address AS block_device_bus_address,
+                    first_value(bdld.name) OVER bdld_first AS block_device_link
+    FROM            storage_volume_attachment sva
+    JOIN            storage_volume sv ON sv.uuid=sva.storage_volume_uuid
+    LEFT JOIN       block_device bd ON bd.uuid=sva.block_device_uuid
+    LEFT JOIN       block_device_link_device bdld ON bd.uuid=bdld.block_device_uuid
+    WHERE           sva.uuid = $volumeAttachmentUUID.uuid
+    WINDOW          bdld_first AS (PARTITION BY bdld.block_device_uuid ORDER BY bdld.name)
```

Comment:
I'd prefer this to be done with an inline view for `SELECT MIN() ... GROUP BY ...`. This is clearer to me that you only want one name matched.

But this poses more questions for me. Please:
- Add a comment to this query as to why you only care about one name.
- Add a comment to the definition of `block_device_link_device` as to its purpose.

Reply:
Good points. I will rework this with new block device work.
We will have some business logic for ranking dev links to select the most appropriate one.